common/lib/dependabot/git_commit_checker.rb

```diff
@@ -86,20 +86,32 @@ def head_commit_for_current_branch
       raise Dependabot::GitDependencyReferenceNotFound, dependency.name
     end
 
-    # rubocop:disable Metrics/PerceivedComplexity
-    # rubocop:disable Metrics/AbcSize
-    def local_tag_for_latest_version
-      tags =
-        local_tags.
-        select { |t| version_tag?(t.name) && matches_existing_prefix?(t.name) }
-      filtered = tags.
-                 reject { |t| tag_included_in_ignore_requirements?(t) }
-      if @raise_on_ignored && filter_lower_versions(filtered).empty? && filter_lower_versions(tags).any?
-        raise Dependabot::AllVersionsIgnored
+    def local_tags_for_latest_version_commit_sha
+      tags = allowed_tags
+      max_tag = tags.
+                max_by do |t|
+        version = t.name.match(VERSION_REGEX).named_captures.
+                  fetch("version")
+        version_class.new(version)
       end
```

Comment:
This seems to be a duplication of lines 114-119, is it maybe worth extracting this to a method?

It took me a few minutes for the need to do the transform from the version back to the tags ( which are versions effectively and likely why I did a double-take 😅  ) on the commit sha to click, isolating this as a method might allow us to make it a little more expressive.

Reply:
I tried to improve this in https://github.com/dependabot/dependabot-core/pull/4953/commits/9d1dafbd16fe3eae46a447d007a6f3747814a1ff but open to any other naming suggestions!